src/compliment/utils.clj

```diff
@@ -19,6 +19,20 @@
 Note that should always have the same value, regardless of OS."
   "/")
 
+(defn split-by-leading-literals
+  "Meant for symbol-strings that might have leading @, #', or '.
+
+  Examples:
+  \"@some-atom\" => '(\"@\" \"some-atom\")
+  \"@#'a\" => '(\"@#'\" \"a\")
+  \"#'some.ns/some-var\" => '(\"#'\" \"some.ns/some-var\")
+
+  \" @wont-work\" => '(nil \" @wont-work\")
+  \"nothing-todo\" => '(nil \"nothing-todo\")
+  "
+  [symbol-str]
+  (rest (re-matches #"(@{0,2}#'|'|@)?(.+)" symbol-str)))
```

Comment:
`next` would work better here because `(next nil) => nil` but `(rest nil) => ()`.

Reply:
OTOH, it can't really not match (except on an empty string), so it doesn't matter.